### ISSUE TITLE
Feature request: New starting point for placement after shift left click

### ISSUE THREAD

**Issue by MarcGamesons** [AkkaWakka/MouseBits#3]
Right now if you place the first object your mod will lock that as the starting point for all other placements after that for the chosen axis.

Would it be possible to update the position, let's say on a shift left click, so that the first object placed on that position will override the previous position and use that as the new starting point for the selected axis?

**Comment by AkkaWakka** [comment]
I've added a hotkey (default is Left Shift) that resets the base.

The issue I ran into however is that shift + left click is used to place ghosts by default.  I'm finding that I'm now inadvertently placing ghosts everywhere.  Not the biggest problem, but not ideal.

Ideally, I'd have liked to be able to Alt + click (or something similar) to place an item with a new base, but for some reason the click/place event gets swallowed by the reset base event.

I definitely think its a good idea, but I'm not sure I like the current default key, or the way it's implemented yet.  I'll keep poking at it, and I'd appreciate any more feedback you might have about how it works/feels.  Would you prefer a press + click over just a press hotkey?

**Comment by MarcGamesons** [comment]
Hmm to be honest I didn't know that shift + left click is already an ingame hotkey, because I didn't use blueprints so far. Didn't finish the game yet, never build the roboports.

I only suggested shift + left click because all your other hotkeys are shift + something. Maybe we/you could use ctrl instead for all your hotkeys? So that they are still consistent, I don't know?

**Comment by AkkaWakka** [comment]
I've ended up sticking with left shift.  Ctrl was a little uncomfortable, and after playing for a while I found I got used to just tapping shift for a reset rather than holding it down.  Otherwise it doesn't seem to be interfering with much, except ghost placement, but only when you let go of shift and press it again, so click drag ghost entities isn't really affected.

I'm going to close this, but feel free to open another issue if you feel something else would be better.
